.travis.yml

```diff
@@ -39,42 +40,60 @@ matrix:
     env: INSTALL_DEPENDS="pip==10.0.1 setuptools==30.3.0"
 
 before_install:
+  - python -m pip install --upgrade pip virtualenv
+  - virtualenv --python=python /tmp/venv
+  - source /tmp/venv/bin/activate
+  - python --version
+  - pip --version
   - python -m pip install --upgrade $INSTALL_DEPENDS
-  - pip install "flake8<3.0" flake8-putty
+  - |
+    if [ "$INSTALL_TYPE" == "pip" ]; then
+        PACKAGE="."
+    elif [ "$INSTALL_TYPE" == "sdist" ]; then
+        python setup.py sdist
+        PACKAGE="$( ls dist/*.tar.gz )"
+    elif [ "$INSTALL_TYPE" == "wheel" ]; then
+        python setup.py bdist_wheel
+        PACKAGE="$( ls dist/*.whl )"
+    fi
```

Comment:
Split out packaging from installation just to make the `install:` section technically accurate. Feel like it's a little cleaner, conceptually.